sdk/eventhub/Azure.Messaging.EventHubs/assets/create-service-principal.ps1

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+  .SYNOPSIS
```

Comment:
I would scope this file to `identity-samples-azure-setup` and include creating the resource group and Event Hubs resources as well.   That would give folks looking to run identity samples a single artifact to use rather than asking them to run the ARM template and then this script.   

We'll want to ensure that it is idempotent in the same manner as the live tests setup; it'll use what you tell it should it already exist and create it if it does not.  We probably also want to limit the role grant to a specific Event Hubs namespace, in order to apply the principle of least privilege should they be using a resource group that they do not own.

Reply:
> We probably also want to limit the role grant to a specific Event Hubs namespace, in order to apply the principle of least privilege should they be using a resource group that they do not own.

I structured role assignments as follow:
- from ```live-tests-azure-setup.ps1``` a single principal will have the ```Contributor``` role and the ```Azure Event Hubs Data Owner``` role. The ```Contributor``` role is needed for the tests to create and configure a namespace and will be assigned to the resource group. The  ```Azure Event Hubs Data Owner``` is needed to produce and consume events in ```Sample13``` and will be assigned to the resource group. Enforcing the principle of least privilege in this instance may not be trivial as the namespace will be created from the tests hence not yet available. We may overcome this by assigning the role from the tests using C#, as you find more convenient.
- from ```identity-samples-azure-setup.ps1``` a principal will be created with only the ```Azure Event Hubs Data Owner``` role assigned to the named namespace. In this instance, being the namespace created by the script will be easy to enforce the principle of least privilege.
- I wonder what would you think about printing out the connection string and the Event Hub name at the end of the script, to keep the experience consistent with what one would have when running the ARM template?

Reply:
> * from `live-tests-azure-setup.ps1` a single principal will have the `Contributor` role and the `Azure Event Hubs Data Owner` role. The `Contributor` role is needed for the tests to create and configure a namespace and will be assigned to the resource group. The  `Azure Event Hubs Data Owner` is needed to produce and consume events in `Sample13` and will be assigned to the resource group. Enforcing the principle of least privilege in this instance may not be trivial as the namespace will be created from the tests hence not yet available. 

I think this is perfect.  The principal used by the tests is trusted and already assumes ownership over the resources within the designated Resource Group, I don't see a benefit to assigning to each EH namespace we create... so long as the `Event Hubs Data Owner` role is inherited.  If that turns out not to be the case, we'll have to do the role assignment with the management libraries within `EventHubScope` as we create them.

Reply:
> I wonder what would you think about printing out the connection string and the Event Hub name at the end of the script, to keep the experience consistent with what one would have when running the ARM template?

Great idea!   I saw an email come through yesterday where you proposed spitting things out formatted like the command line, with all the different arguments specified.   I think that was a fantastic idea, and really liked that.